torchserve-vicuna/langchain_demo_vicuna_torchserve_grpc.py

```diff
@@ -0,0 +1,156 @@
+import re
+import time
+
+import gradio as gr
+from langchain import PromptTemplate, LLMChain
+from langchain.llms.base import LLM
+from langchain.memory import ConversationBufferWindowMemory
+from typing import Iterable
+from gradio.themes.base import Base
+from gradio.themes.utils import colors, fonts, sizes
+import time
+
+import grpc
+import inference_pb2
+import inference_pb2_grpc
+import management_pb2
+import management_pb2_grpc
+
+GRPC_URL="localhost:7070"
+MODEL_NAME = "vicuna-13b"
+responses = None
+
+def get_inference_stub():
+    channel = grpc.insecure_channel("")
+    stub = inference_pb2_grpc.InferenceAPIsServiceStub(channel)
+    return stub
+
+def infer_stream(stub, model_name, model_input):
+    global responses
+    input_data = {"data": bytes(model_input, 'utf-8')}
+
+    responses = stub.StreamPredictions(
+        inference_pb2.PredictionsRequest(model_name=model_name, input=input_data)
+    )
+    return responses
+
+class CustomLLM(LLM):
+    model_name = "custom_model"
+
+    def _call(self, prompt, stop=None) -> str:        
+        infer_stream(get_inference_stub(), MODEL_NAME, prompt)
+        return ""
+
+    @property
+    def _identifying_params(self):
+        return {"name_of_model": self.model_name}
+
+    @property
+    def _llm_type(self) -> str:
+        return "custom"
+
+
+llm = CustomLLM()
```

Comment:
Why do we have this code for LLM outside of any function? Please structure the code into properly